components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -54,16 +115,14 @@ public class SubscriptionServices : ISubscriptionManager, IDisposable {
 
         /// <inheritdoc/>
         public void Dispose() {
+            _timer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan);
```

Comment:
Of course there are still issues as you can see here - the timer task above might just be executing while we run this code.   Hence trying to catch whatever we can.   But we need a better async timer task abstraction going forward... - no action at this point.